python/unityagents/environment.py

```diff
@@ -278,32 +317,52 @@ def step(self, action, memory=None, value=None):
         :param value: Value estimate to send to environment for visualization. Can be a scalar or vector of float(s).
         :return: A Data structure corresponding to the new state of the environment.
         """
+        action = {} if action is None else action
         memory = {} if memory is None else memory
         value = {} if value is None else value
         if self._loaded and not self._global_done and self._global_done is not None:
             if isinstance(action, (int, np.int_, float, np.float_, list, np.ndarray)):
-                if self._num_brains > 1:
+                if self._num_external_brains == 1:
```

Comment:
I tried to run an environment built without broadcast support, and got an error because there was no `externalBrains` dictionary key. We should gracefully use the old approach for environments built without it.